PublicGitArchive/pga/cmd/cache.go

```diff
@@ -66,11 +87,8 @@ func getIndex() (io.ReadCloser, error) {
 	dest := localFS(filepath.Join(usr.HomeDir, ".pga"))
 	source := urlFS(indexURL)
 
-	ok, err := copy(dest, source, indexName)
-	if !ok {
+	if err := updateCache(dest, source, indexName); err != nil {
```

Comment:
From the experience running a workshop with people new to src-d stack, one common pitfall I noticed (2 out of 10 new users) is that after running `pga` for the first time, users do not realize it's actually downloading an index, as there is no CLI feedback and it takes some time - and just perceive it as hang and do `Ctrl+C`.

On the next run, `pga list` does not work for them, as local index is corrupted.

Ofcourse having index md5 will catch that later in #57 , but while we are here, it could be easy to mitigate this type of failure by just printing some CLI output like "Index downloading in progress, please wait."

For sure, this could be done in subesequent PRs, but with the design proposed here `updateCache` already has debug logging built-in but it is used for both, index and .siva files download, while the CLI output is desired only for the index file.

Wondering what would be the best way to do that? Pass `isIndex` to `updateCache` and conditionally logrus.Info from there or is there a better way?

Reply:
Wait, that's exactly what the comment above does, right?
In case where the index is non existent or not up to date we warn that we're downloading a new version.